app/util/Tracing.js

```diff
@@ -0,0 +1,331 @@
+        },
+
+        /**
+         * Finds the closest coordinate to a point.
+         *
+         * @param {ol.coordinate.Coordinate[]} coordinateArray An array of coordinates
+         * @param {ol.coordinate.Coordinate} pointCoordinate The point coordinate
+         *
+         * @returns {ol.coordinate.Coordinate} The found coordinate
+         */
+        getClosestCoordinateToPoint: function (coordinateArray, pointCoordinate) {
+            var staticMe = CpsiMapview.util.Tracing;
+
+            var found, length;
+            Ext.each(coordinateArray, function (c) {
+                var tmpLength = staticMe.computeLength(c, pointCoordinate);
+                if (!length || tmpLength < length) {
+                    length = tmpLength;
+                    found = c;
+                }
+            });
+            return found;
+        },
+
+        /**
+         * Checks if coordinate c is on the [a, b] segment.
+         *
+         * Taken from the official OpenLayers example:
+         * https://openlayers.org/en/latest/examples/tracing.html
+         *
+         * @param {ol.coordinate.Coordinate} c The coordinate to check
+         * @param {ol.coordinate.Coordinate} a The start coordinate of the segment
+         * @param {ol.coordinate.Coordinate} b The end coordinate of the segment
+         *
+         * @returns {Boolean} If coordinate c is on the [a, b] segment
+         */
+        coordIsOnSegment: function (c, a, b) {
+            var staticMe = CpsiMapview.util.Tracing;
+
+            var lengthAc = staticMe.computeLength(a, c);
+            var lengthAb = staticMe.computeLength(a, b);
+            var dot =
+                ((c[0] - a[0]) * (b[0] - a[0]) + (c[1] - a[1]) * (b[1] - a[1])) / lengthAb;
+            return Math.abs(lengthAc - dot) < 1e-6 && lengthAc < lengthAb;
+        },
+
+        /**
+         * Concatenate two coordinate arrays if they are touching at the startpoint or endpoint.
+         * It takes the direction of the arrays into account.
+         *
+         * Return empty array otherwise.
+         *
+         * @param {ol.coordinate.Coordinate[]} aLineCoords The first coordinate array
+         * @param {ol.coordinate.Coordinate[]} bLineCoords The second coordinate array
+         * @returns {ol.coordinate.Coordinate[]} The combined coordinate array if input arrays are touching, empty array otherwise
+         */
+        concatLineCoords: function (aLineCoords, bLineCoords) {
+            var aFirst = aLineCoords[0] || [];
+            var aLast = aLineCoords[aLineCoords.length - 1] || [];
+
+            var bFirst = bLineCoords[0] || [];
+            var bLast = bLineCoords[bLineCoords.length - 1] || [];
+
+            var lastFirst = Ext.Array.equals(aLast, bFirst);
+            var lastLast = Ext.Array.equals(aLast, bLast);
+            var firstFirst = Ext.Array.equals(aFirst, bFirst);
+            var firstLast = Ext.Array.equals(aFirst, bLast);
+
+            if (lastFirst) {
```

Comment:
I'm not sure, if I like empty if blocks. Maybe it's cleaner to replace the last `else` with an `else if(!lastFirst)` or something alike

Reply:
done with [62affc3](https://github.com/compassinformatics/cpsi-mapview/pull/586/commits/62affc3d231bf332821c2547844d05685aa2359f)